File: 3819-rotate-non-negative-elements/3819-rotate-non-negative-elements.java
class Solution {
    public int[] rotateElements(int[] nums, int k) {
        int n = nums.length, m = 0;
        int[] a = new int[n];

        for (int num : nums) {
            if (num >= 0)
                a[m++] = num;
        }
        if (m == 0)
            return nums;
        k %= m;
        for (int i = 0, j = 0; i < n; i++) {
            if (nums[i] >= 0)
                nums[i] = a[(j++ + k) % m];
        }
        return nums;
    }
}

File: 3819-rotate-non-negative-elements/3819-rotate-non-negative-elements.py
class Solution:
    def rotateElements(self, nums: List[int], k: int) -> List[int]:
        m, n = 0, len(nums)
        a = [0] * n

        for num in nums:
            if num >= 0:
                a[m] = num
                m += 1
        
        if m == 0:
            return nums
        
        k %= m
        i, j = 0, 0

        while i < n:
            if nums[i] >= 0:
                nums[i] = a[(j + k) % m]
                j += 1
            i += 1
        return nums